File: src/analyzer/format-var-diagnostic.ts
import type { Diagnostic } from "../ir/diagnostic/diagnostic.js";

// Returns the lines a var-detected diagnostic consists of, sharing the
// exact wording between the stderr emitter and the markdown Notice
// section so both stay in lock-step.
export function formatVarDiagnostic(diagnostic: Diagnostic): string[] {
  const { line, column } = diagnostic.span;
  return [`uns: warning: L${line}:${column}: ${diagnostic.message}`];
}

File: crates/unsnarl-analyzer/src/format_var_diagnostic.rs
//! Lines the var-detected diagnostic consists of.
//!
//! The same wording is shared between the stderr emitter and the
//! markdown Notice section so both stay in lock-step.

use unsnarl_ir::diagnostic::Diagnostic;

pub fn format_var_diagnostic(diagnostic: &Diagnostic) -> Vec<String> {
    vec![format!(
        "uns: warning: L{}:{}: {}",
        diagnostic.span.line.0, diagnostic.span.column.0, diagnostic.message
    )]
}

#[cfg(test)]
#[path = "format_var_diagnostic_test.rs"]
mod format_var_diagnostic_test;
